src/main/java/org/zalando/nakadi/service/subscription/SubscriptionTimeLagService.java

```diff
@@ -82,21 +83,12 @@ public SubscriptionTimeLagService(final TimelineService timelineService,
             for (final EventTypePartition partition : futureTimeLags.keySet()) {
                 timeLags.put(partition, futureTimeLags.get(partition).get());
             }
-            return timeLags;
-        } catch (final RejectedExecutionException e) {
-            throw new LimitReachedException("Time lag statistics thread pool exhausted", e);
-        } catch (final TimeoutException e) {
-            throw new TimeLagStatsTimeoutException("Timeout exceeded for time lag statistics", e);
-        } catch (final ExecutionException e) {
-            if (e.getCause() instanceof NakadiBaseException) {
```

Comment:
is it possible that among the possible `NakadiBaseException`s there would be an authorization exception? I'm not sure if authorization is checked earlier in the code path, but if that's the case, we should handle this one specially. Could you please check that?

Reply:
Forget about this because authorization doesn't protect stats.